.github/workflows/MarkdownLinksCheck.yml

```diff
@@ -2,18 +2,21 @@ name: Check Markdown links
 
 on: 
   pull_request:
-    branches:
-      - master
+  push:
 
+permissions:
+  contents: read
+  
 jobs:
   markdown-link-check:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
       with:
         ref: ${{ github.event.pull_request.head.sha }}
-    - uses: gaurav-nelson/github-action-markdown-link-check@2a60e0fe41b5361f446ccace6621a1a2a5c324cf
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
```

Comment:
Now with read-only permissions it's ok for me to use just a released tag version and not the SHA of the commit :)